gammapy/modeling/models/core.py

```diff
@@ -411,16 +413,32 @@ def names(self):
         return [m.name for m in self._models]
 
     @classmethod
-    def read(cls, filename):
-        """Read from YAML file."""
+    def read(cls, filename, checksum=False):
+        """Read from YAML file.
+
+        Parameters
+        ----------
+        filename : str
+            input filename
+        checksum : bool
```

Comment:
Astropy has a more options for fits: https://docs.astropy.org/en/stable/io/fits/api/verification.html

I think `exception`, `ignore` and `warn` would all make sense here as well.

Reply:
As far as I understand, the verification scheme you mention concerns the compliance with the FITS standard not the data integrity. 
I have seen only this regarding data integrity: https://docs.astropy.org/en/stable/io/fits/usage/verification.html#verification-using-the-fits-checksum-keyword-convention
The behavior is to raise an  `AstropyUserWarning` in case checksum verification failed.
I have tried to do the same (except we don't have a `GammapyUserWarning`, but we could create one).

Reply:
Yes, you are right. Thanks for pointing it out, please ignore the previous comment.